src/libs/map-engine.ts

```diff
@@ -9,7 +9,7 @@ const baseMarkerDimension: number = 10;
 
 export default class MapEngine {
     private map: google.maps.Map;
-    private popup?: google.maps.InfoWindow;
+    private icaoNumberToPopup: { icaoNumber?: string, popup?: google.maps.InfoWindow } = {};
```

Comment:
I don't like that both the fields are nullable, but it's probably harder to check that the field is not null on each call, so ok.